TidepoolOnboarding/Views/Sections/YourSettingsViews.swift

```diff
@@ -22,8 +24,12 @@ struct YourSettingsNavigationButton: View {
 
     @ViewBuilder
     private var destination: some View {
-        if !tidepoolServiceOnboarded {
+        if !tidepoolServiceOnboarded, !deviceValidated {
```

Comment:
[nit] This is a total non-blocking nit, but all these `Bool`s seem to indicate that maybe an `enum` might be called for instead?  Just a thought...

Reply:
True. I'm not sure it would simplify things that much because it would just move all of the if/else-if/else logic into the `action` function.